### ISSUE TITLE
Enigma: Importing GPG keys - Failed to send the file

### ISSUE THREAD

**Issue by WisdomEye** [roundcube/roundcubemail#7160]
On my installation I can:

1.  Generate keys from UI
2.  Sign mail using the Generated keys from UI
3.  Encrypt the mail using the Generated keys from UI
4.  Export the Keys Generated from UI
5.  Delete the keys Generated from UI

System:

FreeBSD: 12.0
RoundCube: 1.4.1_1,1
PHP: 7.3.13
Apache: 2.4.41

PS.:  Everything compiled from /usr/ports

The import fails in both situation:

- Importing back the key generated from Roundcube UI
- Importing any other external key

Both After select the file on Upload File Path fails after click **Import** with message:

Fail to send the file

After clear the **roundcube/log/enigma.log** and and try import again the output is:

[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: /usr/local/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options --always-trust --ignore-time-conflict --ignore-valid-from --homedir '/var/lib/gnupg/enigma/private@domain.com' --version
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: BEGIN PROCESSING
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: selecting streams
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => got 1
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: GPG output stream ready for reading
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG output
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => read 625 bytes
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: selecting streams
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => got 3
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: GPG output stream ready for reading
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG output
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => read 0 bytes
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: GPG error stream ready for reading
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG error
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => read 0 bytes
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: GPG status stream ready for reading
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG status
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: => read 0 bytes
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: END PROCESSING
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: CLOSING GPG SUBPROCESS
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: USING GPG 2.2.19 with PHP 7.3.13
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: /usr/local/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options --no-permission-warning --exit-on-status-write-error --trust-model always --pinentry-mode loopback --ignore-time-conflict --ignore-valid-from --digest-algo 'SHA512' --s2k-digest-algo 'SHA512' --cipher-algo 'AES256' --s2k-cipher-algo 'AES256' --with-colons --with-fingerprint --with-fingerprint --fixed-list-mode --homedir '/var/lib/gnupg/enigma/private@domain.com' --list-secret-keys
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: BEGIN PROCESSING
[09-Jan-2020 20:24:23 -0300]: <xRwtUAo5> GPG: selecting streams
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => got 3
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: GPG output stream ready for reading
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG output
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => read 0 bytes
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: GPG error stream ready for reading
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG error
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => read 0 bytes
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: GPG status stream ready for reading
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG status
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => read 0 bytes
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: END PROCESSING
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: CLOSING GPG SUBPROCESS
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: USING GPG 2.2.19 with PHP 7.3.13
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: /usr/local/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options --no-permission-warning --exit-on-status-write-error --trust-model always --pinentry-mode loopback --ignore-time-conflict --ignore-valid-from --digest-algo 'SHA512' --s2k-digest-algo 'SHA512' --cipher-algo 'AES256' --s2k-cipher-algo 'AES256' --with-colons --with-fingerprint --with-fingerprint --fixed-list-mode --homedir '/var/lib/gnupg/enigma/private@domain.com' --list-public-keys
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: BEGIN PROCESSING
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: selecting streams
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => got 3
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: GPG output stream ready for reading
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG output
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => read 18 bytes
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: GPG error stream ready for reading
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG error
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => read 0 bytes
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: GPG status stream ready for reading
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG status
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => read 0 bytes
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: selecting streams
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => got 1
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: GPG output stream ready for reading
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => about to read 65536 bytes from GPG output
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: => read 0 bytes
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: END PROCESSING
[09-Jan-2020 20:24:24 -0300]: <xRwtUAo5> GPG: CLOSING GPG SUBPROCESS
`

If I do an import from external key using the terminal, the key show at UI, but, after click to send the mail using the manual imported key there's an error about cannot find the specified key...

If I do an import from external key using the terminal, I cannot delete the key using the interface, it says cannot find the specified file to delete.

On **roundcube/log/error.log** there no error.

**Comment by alecpl** [comment]
Works for me. The log as well as "Fail to send the file" suggests some generic upload issue. Could you provide a screenshot with exact error message? Maybe some error in browser console or http server logs?

**Comment by WisdomEye** [comment]
Sorry for the delay on this issue, and thanks for the answer!!!

You are right, its an problem with permissions.

After your hint, I have tried put an attachment on mail and got same error about:

`File upload failed.`

(Was using the UI on language Portuguese Brazil, and did an wrong translation, sorry).

I am now trying figure out where is the problem.

The file **~/config/config.inc.php** with the options:

```
$config['debug_level'] = 4;
$config['sql_debug'] = true;
```

Don't add anything on logs at **~logs/errors.log**.

Checking the system logs at **/var/log/httpd-error.log** there's nothing too.

Then I have verified the apache vhost and there's an custom log on that (finally!!!).

All updates do the error:

`proxy_fcgi:error ... Unknow: file created in the system's temporary directory in Unknow on line 0PHP message: PHP Warning: File upload error - unable to create a temporary file in Unknown on line 0`

Have tried **chmod 777** on **~/temp**, does not work.
Owner is same from webserver **www:www** (same **php-fpm** use too).

Have tried also use **global temp** file **shared in php** without success too.

I will do more tests, and back later to  update this.

I am doing some tests on specific php-fpm pool I have did for the roundcube, since other's sites are working well (wordpress and others - each with specific php-fpm pool).
